build_projects/dotnet-cli-build/CompileTargets.cs

```diff
@@ -261,6 +281,69 @@ private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli d
             File.WriteAllText(Path.Combine(sdkOutputDirectory, ".version"), content);
 
             return c.Success();
+        }        
+
+        private static void GenerateNuGetPackagesArchive(
+            BuildTargetContext c,
+            DotNetCli dotnet,
+            string sdkOutputDirectory)
+        {
+            var nuGetPackagesArchiveProject = Path.Combine(Dirs.Intermediate, "NuGetPackagesArchiveProject");
+            var nuGetPackagesArchiveFolder = Path.Combine(Dirs.Intermediate, "nuGetPackagesArchiveFolder");
+
+            RestoreNuGetPackagesArchive(dotnet, nuGetPackagesArchiveProject, nuGetPackagesArchiveFolder);
+
+            CompressNuGetPackagesArchive(c, dotnet, nuGetPackagesArchiveFolder, sdkOutputDirectory);
+        }
+
+        private static void RestoreNuGetPackagesArchive(
+            DotNetCli dotnet,
+            string nuGetPackagesArchiveProject,
+            string nuGetPackagesArchiveFolder)
+        {
+            Rmdir(nuGetPackagesArchiveProject);
+            Mkdirp(nuGetPackagesArchiveProject);
+
+            Rmdir(nuGetPackagesArchiveFolder);
+            Mkdirp(nuGetPackagesArchiveFolder);
+
+            dotnet.New()
+                .WorkingDirectory(nuGetPackagesArchiveProject)
+                .Execute()
+                .EnsureSuccessful();
+
+            dotnet.Restore("--packages", nuGetPackagesArchiveFolder)
+                .WorkingDirectory(nuGetPackagesArchiveProject)
+                .Execute()
+                .EnsureSuccessful();
+        }
+
+        private static void CompressNuGetPackagesArchive(
+            BuildTargetContext c,
+            DotNetCli dotnet,
+            string nuGetPackagesArchiveFolder,
+            string sdkOutputDirectory)
+        {
+            var configuration = c.BuildContext.Get<string>("Configuration");
+            var archiver = Path.Combine(Dirs.Output, "tools", $"Archiver{Constants.ExeSuffix}");
+            var intermediateArchive = Path.Combine(Dirs.Intermediate, "nuGetPackagesArchive.lzma");
+            var finalArchive = Path.Combine(sdkOutputDirectory, "nuGetPackagesArchive.lzma");
+
+            Rm(intermediateArchive);
+            Rm($"{intermediateArchive}.zip");
+
+            c.Info("Publishing Archiver");
+            dotnet.Publish("--output", Path.Combine(Dirs.Output, "tools"), "--configuration", configuration)
+                .WorkingDirectory(Path.Combine(c.BuildContext.BuildDirectory, "tools", "Archiver"))
```

Comment:
use Dirs.RepoRoot instead of this BuildDirectory. I'm on a quest to remove these BuildDirectory's all over the place